### ISSUE TITLE
text overlap

### ISSUE THREAD

**Issue by luke-hagar-sp** [josdejong/svelte-jsoneditor#206]
Hey Just a heads up, with the static width I'm getting some overlap, so I'm just throwing a small adjustment in my project
![image](https://user-images.githubusercontent.com/98849695/206776721-a94c9820-fdfc-4398-9518-7cf5974773b1.png)

```css
.jse-group-button {
	@apply !w-auto !px-1;
}
```

**Comment by luke-hagar-sp** [comment]
Here is after, not a big deal at all
![image](https://user-images.githubusercontent.com/98849695/206776979-26958fbf-fe59-40bc-a7b8-5d2f3f1121d8.png)

**Comment by josdejong** [comment]
Thanks Luke for your input. How can I reproduce this issue? Does your project have specific CSS causing this issue?

**Comment by luke-hagar-sp** [comment]
Hey,

Yes, this is absolutely a side effect of global styles on my side, I’m using the styling system and theme formats from a component library I contribute a little to here and there, link is here if you want to take a look. https://www.skeleton.dev/guides/themes


Luke Hagar
Cloud Support Engineer
***@***.*** ***@***.***>
Join the #SailPointCrew <https://www.sailpoint.com/company/careers/>

[https://i.xink.io/Images/Get/N9791/s18.png]

From: Jos de Jong ***@***.***>
Date: Monday, December 12, 2022 at 6:58 AM
To: josdejong/svelte-jsoneditor ***@***.***>
Cc: Luke Hagar ***@***.***>, Author ***@***.***>
Subject: Re: [josdejong/svelte-jsoneditor] text overlap (Issue #206)
Thanks Luke for your input. How can I reproduce this issue? Does your project have specific CSS causing this issue? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread. Message
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

Thanks Luke for your input. How can I reproduce this issue? Does your project have specific CSS causing this issue?

**Comment by josdejong** [comment]
Ah ok, We we can make the library more robust that would be nice. 

Is it possible to write fix with a CSS rule that does not use framework specific variables like `w-auto` and `px-1`?

**Comment by luke-hagar-sp** [comment]
Absolutely,

Those are tailwind classes, so they always equate to standard CSS values.

w-auto is equal to:
width: auto;

And px-1 is equal to:
padding-left: 0.25rem;
padding-right: 0.25rem;

I love tailwind personally; it just makes styling easier for me.
In general, for those kinds of things, I would avoid hardcoded h/w values, if you can base things like button sizing on padding or margin instead it usually plays better with different text sizes.

Really loving what you have built here.

Thank you,

Luke Hagar
Cloud Support Engineer
***@***.*** ***@***.***>
Join the #SailPointCrew <https://www.sailpoint.com/company/careers/>

[https://i.xink.io/Images/Get/N9791/s18.png]

From: Jos de Jong ***@***.***>
Date: Monday, December 12, 2022 at 7:49 AM
To: josdejong/svelte-jsoneditor ***@***.***>
Cc: Luke Hagar ***@***.***>, Author ***@***.***>
Subject: Re: [josdejong/svelte-jsoneditor] text overlap (Issue #206)
Ah ok, We we can make the library more robust that would be nice. Is it possible to write fix with a CSS rule that does not use framework specific variables like w-auto and px-1? — Reply to this email directly, view it on GitHub, or unsubscribe. You
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

Ah ok, We we can make the library more robust that would be nice.

Is it possible to write fix with a CSS rule that does not use framework specific variables like w-auto and px-1?

**Comment by josdejong** [comment]
Ahh, now I see I gave the buttons a fixed width of 40px for some reason, that should indeed be auto. Fixed via https://github.com/josdejong/svelte-jsoneditor/commit/8e0cda3c1c91e97b1c41ff28c5b0ac80cf8c26ab. 

The editor uses a configurable padding (defaulting to `10px`) in many places. It would be interesting to see if we can make the padding relative to the font size in more places, but better to address that in a separate refactoring.
